community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/CypherException.scala

```diff
@@ -146,3 +146,19 @@ object ExhaustiveShortestPathForbiddenException {
        |start filtering.""".stripMargin
 }
 
+class ShortestPathCommonEndNodesForbiddenException extends CypherExecutionException(
+  ShortestPathCommonEndNodesForbiddenException.ERROR_MSG, null) {
+  override def mapToPublic[T <: Throwable](mapper: MapToPublicExceptions[T]): T =
+    mapper.shortestPathCommonEndNodesForbiddenException(message, this)
+}
+
+object ShortestPathCommonEndNodesForbiddenException {
+  val ERROR_MSG: String =
+    s"""The shortest path algorithm does not work when the start and end nodes are the same. This can happen if you
+        |perform a shortestPath search after a cartesian product that might have the same start and end nodes for some
+        |of the rows passed to shortestPath. If would rather not experience this exception, and can accept the
+        |possibility of missing results for those rows, disable this in the Neo4j configuration. If you cannot accept
```

Comment:
Please mention the configuration key when referring here: `cypher.forbid_shortestpath_common_nodes`, and that setting it to `false` will disable the exception.

Reply:
my neo4j is 3.5.4  and i don't have tthis line  in  my conf file

Reply:
It is in 3.5, see https://github.com/neo4j/neo4j/blob/3.5/community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java#L227

The fact that it is not shown in the default neo4j.conf file is because only the most commonly edited settings are shown there. 

It is also in the Neo4j documentation at https://neo4j.com/docs/operations-manual/current/reference/configuration-settings/#config_cypher.forbid_shortestpath_common_nodes

Reply:
I get it .And I solved this shortestpath problem by adding：`match paths = shortestpath((source)-[*0..2]-(target)) where source<>target` This can return right result.